runtime/src/lib.rs

```diff
@@ -258,11 +269,154 @@ impl balances::Trait for Runtime {
 	type WeightToFee = ConvertInto;
 }
 
+parameter_types! {
+	pub const UncleGenerations: BlockNumber = 5;
+}
+
+impl authorship::Trait for Runtime {
+	type FindAuthor = session::FindAccountFromAuthorIndex<Self, Babe>;
+	type UncleGenerations = UncleGenerations;
+	type FilterUncle = ();
+	type EventHandler = Staking;
+}
+
+type SessionHandlers = (Grandpa, Babe, ImOnline, AuthorityDiscovery);
+
+impl_opaque_keys! {
```

Comment:
session keys struct is already implemented in our runtime you can just add the new one that is `im_online`

Reply:
Changed.